packages/ui-dom-utils/src/isDefinedCustomElement.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * ---
+ * category: utilities/DOM
+ * ---
+ *
+ * Check wheter an element is a defined custom element. (it was defined with `customElements.define`)
+ * @module isCustomElement
+ * @param { Element } el - the element to check
+ * @return { Boolean } wheter the element is a custom element or not
+ */
+function isDefinedCustomElement(el: Element) {
+  return Boolean(
+    el.tagName.includes('-') && customElements.get(el.tagName.toLowerCase())
```

Comment:
I saw that post, and I believe that check is redundant and not necessary since the [CustomElementRegistry](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry) can only contain instances that are defined by the [define](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define) API on window.customElements.

Also the code that the author wrote is wrong, the first check should return true if the node has a `-` in its tagname.